pallets/parachain-staking/src/tests/bond_extra_tests.rs

```diff
@@ -18,6 +18,232 @@ fn to_acc_id(id: u64) -> AccountId {
+        let proof = create_proof_for_signed_bond_extra(sender_nonce, staker, &extra_amount);
+
+        return Box::new(MockCall::ParachainStaking(super::super::Call::<Test>::signed_bond_extra {
+            proof,
+            extra_amount,
+        }))
+    }
+
+    fn create_proof_for_signed_bond_extra(
+        sender_nonce: u64,
+        staker: &Staker,
+        extra_amount: &u128,
+    ) -> Proof<Signature, AccountId> {
+        let data_to_sign = encode_signed_bond_extra_params::<Test>(
+            staker.relayer.clone(),
+            extra_amount,
+            sender_nonce,
+        );
+
+        let signature = sign(&staker.key_pair, &data_to_sign);
+        return build_proof(&staker.account_id, &staker.relayer, signature)
+    }
+
+    #[test]
+    fn succeeds_with_good_parameters() {
+        let collator_1 = to_acc_id(1u64);
+        let collator_2 = to_acc_id(2u64);
+        let staker: Staker = Default::default();
+        let initial_stake = 10;
+        let initial_balance = 10000;
+        ExtBuilder::default()
+            .with_balances(vec![
+                (collator_1, initial_balance),
+                (collator_2, initial_balance),
+                (staker.account_id, initial_balance),
+                (staker.relayer, initial_balance),
+            ])
+            .with_candidates(vec![(collator_1, initial_stake), (collator_2, initial_stake)])
+            .with_nominations(vec![
+                (staker.account_id, collator_1, initial_stake),
+                (staker.account_id, collator_2, initial_stake),
+            ])
+            .build()
+            .execute_with(|| {
+                let collators = ParachainStaking::selected_candidates();
+                let min_user_stake = MinNominationPerCollator::get();
+                let initial_total_stake_on_chain = ParachainStaking::total();
+
+                // Pick an amount that is not perfectly divisible by the number of collators
+                let dust = 1u128;
+                let amount_to_topup = (min_user_stake * 2u128) + dust;
+                let nonce = ParachainStaking::proxy_nonce(staker.account_id);
+                let bond_extra_call = create_call_for_bond_extra(&staker, nonce, amount_to_topup);
+                assert_ok!(AvnProxy::proxy(Origin::signed(staker.relayer), bond_extra_call, None));
+
+                // The staker state has also been updated
+                let staker_state = ParachainStaking::nominator_state(staker.account_id).unwrap();
+                assert_eq!(staker_state.total(), initial_stake * 2 + amount_to_topup);
+
+                // Each collator has been topped up by the expected amount
+                for (index, collator) in collators.into_iter().enumerate() {
+                    // We should have one event per collator. One of the collators gets any
+                    // remaining dust.
+                    let mut topup = min_user_stake;
+                    if index == 1 {
```

Comment:
Is the index 1-based or 0-based? What is the logic of this 1, is it because it is the last?
I think this should be made clearer, it does not explain what the expected functionality (or the logic we baked in) should be.

Reply:
The index is 0 based, and this check is there to find the collator that got the `dust` amount.